### PR TITLE
DM XML: Add device types to script and re-gen

### PR DESCRIPTION
**Pull request by cecille**
see https://github.com/CHIP-Specifications/chip-test-plans/issues/3675

### CODE REVIEW

**Review by plauric — APPROVED** [review]

File: data_model/device_types/BaseDeviceType.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+All company, brand and product names in this document may be trademarks that
+are the sole property of their respective owners.
+
+This notice and disclaimer must be included on all copies of this document.
+
+Connectivity Standards Alliance
+508 Second Street, Suite 206
+Davis, CA 95616, USA
+-->
+<deviceType xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="types types.xsd devicetype devicetype.xsd" id="" name="Base Device Type" revision="2">
```
Comment (ReneJosefsen) [inline data_model/device_types/BaseDeviceType.xml:58]: This seems to be missing the requirement of at least the descriptor cluster, then the others is a bit more complex to get added here 🤔

File: data_model/device_types/section_Application-Device-Types.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0"?>
```
Comment (ReneJosefsen) [inline data_model/device_types/section_Application-Device-Types.xml:1]: Shouldn't these types of files be removed? There is a few so only commented on the first one in the list. 😄

Comment (ReneJosefsen) [reply data_model/device_types/section_Application-Device-Types.xml:1]: Can we have the scraper ignore spec documents that starts with "section_"?

File: scripts/spec_xml/generate_spec_xml.py [hunk 3]
```diff
@@ -94,5 +102,27 @@ def scrape_all_clusters(dir: str, exclude_list: list[str] = []) -> None:
...
+        xml_path = get_xml_path(filename, device_types_output_dir)
+        print(xml_path)
+        cmd = [scraper, 'devicetype', '-c', clusters_output_dir, '-nd', filename, xml_path]
+        if dry_run:
+            print(cmd)
+        else:
+            print(' '.join(cmd))
+            subprocess.run(cmd)
+
+    for filename in glob.glob(f'{device_type_dir}/*.adoc'):
```
Comment (ReneJosefsen) [inline scripts/spec_xml/generate_spec_xml.py:123]: I guess this for loop needs to be adjusted a bit to not include the "sections_" files as well?

File: data_model/device_types/section_Application-Device-Types.xml (hunk 2 above)
Comment (cecille) [reply data_model/device_types/section_Application-Device-Types.xml]: yup, fair - done.

**Comment by cecille** [comment]
I re-ran this to remove the section_ files, and in the process I also updated the scraper and then added a file marker for that. So that's where all the new changes are coming from.

**Review by woody-apple — APPROVED** [review]

**Review by ReneJosefsen — APPROVED** [review]

**Pull request merged**